flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -125,6 +132,8 @@ public JobVertex(String name) {
 	public JobVertex(String name, JobVertexID id) {
 		this.name = name == null ? DEFAULT_NAME : name;
 		this.id = id == null ? new JobVertexID() : id;
+		this.operatorIDs.add(new OperatorID(this.id.getLowerPart(), this.id.getUpperPart()));
+		this.operatorIdsAlternatives.add(null);
```

Comment:
We only need a single alternative ID for each operatorID, the one set using `setUIDHash()`.

For savepoints created with 1.3 other alternatives aren't required since 1.3 doesn't use the old hasher.

1.0-1.2 savepoints are converted to the 1.3 format using the alternative job vertex IDs (see `SavepointV2#convertToOperatorStateSavepointV2`), which is why we can't remove them. After the conversion however the are identical to a 1.3 savepoint.